File: atcoder/tyama_atcodertrickyB2.nim
#[
/usr/bin/env nim c --nimcache:$HOME/.nimcache --opt:speed --run $0 $@;exit
#]#
import strutils,math
proc scanf(formatstr: cstring){.header:"<stdio.h>",varargs.}
proc printf(formatstr: cstring){.header:"<stdio.h>",varargs.}

var T=0
#T=parseInt(readLine(stdin))
scanf("%d",T.addr)
for i in 1..T:
 var a=0.int64
 var b=0.int64
 var c=0.int64
 #var ABC=readLine(stdin).split()
 #a=parseInt(ABC[0]).int64
 #b=parseInt(ABC[1]).int64
 #c=parseInt(ABC[2]).int64
 scanf("%lld%lld%lld",a.addr,b.addr,c.addr)
 if a==0:
  if b==0:
   #echo(if c!=0: "0" else: "3")
   printf(if c!=0: "0\n" else: "3\n")
  else:
   #echo("1 "&(-c.float64/b.float64).formatFloat(ffDecimal,12))
   printf("1 %.12f\n",-c.float64/b.float64)
 else:
  if a<0:
   a = -a
   b = -b
   c = -c
  var d=b.float64*b.float64-4.0*a.float64*c.float64
  if d.abs<1e-9:
   #echo("1 "&(-b.float64/2.0/a.float64).formatFloat(ffDecimal,12))
   printf("1 %.12f\n",-b.float64/2.0/a.float64)
  elif d<0:
   #echo("0")
   printf("0\n")
  else:
   # acknowledgement: http://tricky.contest.atcoder.jp/submissions/120839
   var x=0.0
   var y=0.0
   if b>0:
    x = (-b.float64-sqrt(d))/2/a.float64
    y = c.float64/a.float64/x
   else:
    y = (-b.float64+sqrt(d))/2/a.float64
    x = c.float64/a.float64/y
   #echo("2 "&x.formatFloat(ffDecimal,12)&" "&y.formatFloat(ffDecimal,12))
   printf("2 %.12f %.12f\n",x,y)

File: atcoder/tyama_atcodertrickyB2.swift
//usr/bin/env swift $0 $@;exit
import CoreFoundation
//import Foundation
//Note: String(format:) is not implemented on Linux
var T=0
var a:Int64=0
var b:Int64=0
var c:Int64=0
_=withUnsafePointer(to:&T){withVaList([OpaquePointer($0)]){vscanf("%d",$0)}}
//=Int(readLine()!)!
for var i in 0..<T {
	_=withUnsafePointer(to:&a){withVaList([OpaquePointer($0)]){vscanf("%lld",$0)}}
	_=withUnsafePointer(to:&b){withVaList([OpaquePointer($0)]){vscanf("%lld",$0)}}
	_=withUnsafePointer(to:&c){withVaList([OpaquePointer($0)]){vscanf("%lld",$0)}}
	//var _a=readLine()!.characters.split{$0==" "}
	//a=Int64(String(_a[0]))!
	//b=Int64(String(_a[1]))!
	//c=Int64(String(_a[2]))!
	if a==0 {
		if b==0 {print(c != 0 ? "0" : "3")}
		else {
			_=withVaList([-Double(c)/Double(b)]){vprintf("1 %.12f\n",$0)}
			//print(String(format:"1 %.12f",-Double(c)/Double(b)))
		}
	} else{ 
		if a<0 {
			a = -a
			b = -b
			c = -c
		}
		var d=Double(b)*Double(b)-4.0*Double(a)*Double(c)
		if fabs(d)<1e-9 {
			_=withVaList([-Double(b)/2.0/Double(a)]){vprintf("1 %.12f\n",$0)}
			//print(String(format: "1 %.12f",-Double(b)/2.0/Double(a)))
		}
		else if d<0 {print("0")}
		else{
			// acknowledgement: http://tricky.contest.atcoder.jp/submissions/120839
			var x=0.0,y=0.0;
			if b>0 {
				x = (-Double(b)-sqrt(d))/2/Double(a)
				y = Double(c)/Double(a)/x
			}else{
				y = (-Double(b)+sqrt(d))/2/Double(a)
				x = Double(c)/Double(a)/y
			}
			_=withVaList([x]){vprintf("2 %.12f ",$0)}
			_=withVaList([y]){vprintf("%.12f\n",$0)}
			//print(String(format: "2 %.12f %.12f",x,y))
		}
	}
}
